CHANGELOG.md

```diff
@@ -23,6 +23,92 @@
+* Fix the dispatcher() stuck caused by availablePermitsCh by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/875
+* Fix the Send() stuck caused by callback() not being called by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/880
+* Fix the data race of ackReq.err by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/881
+* Add data url format to read the key file by @nodece in https://github.com/apache/pulsar-client-go/pull/883
+* Prevent consumer panic on de-serializing message if schema not found by @GPrabhudas in https://github.com/apache/pulsar-client-go/pull/886
+* Fix load tls certificate by @nodece in https://github.com/apache/pulsar-client-go/pull/888
+* Fix default retry and dlq topic name as per the doc by @ngoyal16 in https://github.com/apache/pulsar-client-go/pull/891
+* Add NewMessageID() method. by @crossoverJie in https://github.com/apache/pulsar-client-go/pull/893
+* Use protocolbuffers instead of gogo by @nodece in https://github.com/apache/pulsar-client-go/pull/895
+* Fix the compression broken when batching is disabled by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/902
+* Add messageId and topic as props of DLQ message by @GPrabhudas in https://github.com/apache/pulsar-client-go/pull/907
+* Update go version to 1.18 by @pgier in https://github.com/apache/pulsar-client-go/pull/911
+* Move out the auth package from internal by @nodece in https://github.com/apache/pulsar-client-go/pull/914
+* Remove the `clearMessageQueuesCh` in `partitionConsumer.dispatcher()` by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/921
+* Remove the outdated interface description of `SeekByTime` by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/924
+* Handle nil value message correctly in table-view by @Demogorgon314 in https://github.com/apache/pulsar-client-go/pull/930
+* Migrate from the deprecated io/ioutil package by @reugn in https://github.com/apache/pulsar-client-go/pull/942
+* Update the Cobra library to significantly reduce the dependency tree by @reugn in https://github.com/apache/pulsar-client-go/pull/943
+* Remove go1.11 code leftovers by @reugn in https://github.com/apache/pulsar-client-go/pull/946
+* Use pkg.go.dev badge in the readme by @reugn in https://github.com/apache/pulsar-client-go/pull/947
+* Improve test script by @nodece in https://github.com/apache/pulsar-client-go/pull/951
+* Optimize the performance by passing MessageID implementations by pointers by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/968
+* Fix flaky Key_Shared subscription related tests by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/970
+* Refactor the toTrackingMessageID() by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/972
+* Prevent rpc client panic on rpc response if `ProducerReady` is nil by @sekfung in https://github.com/apache/pulsar-client-go/pull/973
```

Comment:
```suggestion
* Prevent RPC client panic on RPC response if `ProducerReady` is nil by @sekfung in https://github.com/apache/pulsar-client-go/pull/973
```
BTW, is 'nil' a correct value or short for something else?

Reply:
Yes, it represents `null` in go.